src/systems/diffeqs/diffeqsystem.jl

```
@@ -16,51 +16,53 @@ end
 
 
 struct DiffEq  # dⁿx/dtⁿ = rhs
-    x::Expression
-    t::Variable
+    x::Variable
```

This makes invalid states unrepresentable, leading to less bugs/confusion. In `to_diffeq`, we enforce the necessary invariants about each `DiffEq` and store the independent variable `t` externally (in the `DiffEqSystem`), guaranteeing that all `t`s are equal after construction.

---

the LHS isn't necessarily a single variable. If we want to enforce this, we should be strict with the naming and call it a mass-matrix free ODE.

---

Hmm. Does this mean that our existing functionalities on `DiffEqSystem`s support things like `D(sin(x)) ~ ...`? And what about `2D(sin(x)) + D(y)^2 ~ ...`? I assumed we wanted things of the form `[D(x) ~ ..., D(y) ~ ...]` based on the test cases, and thus tried to mirror that with the shape of the data itself, but if that's not the case I can easily switch it.

The purpose behind this change was to ensure that the invariants implicitly assumed in other functions (e.g. `first_order_transform`) hold for any constructed system, rather than having that assumption be false and throwing a runtime error.

---

>And what about 2D(sin(x)) + D(y)^2 ~ ...?

That's a DAE, and if it's linear in the derivatives, it's a mass matrix ODE. We want to go there next. Since we're going to be hitting this issue soon, we should start thinking about how to cleanly handle the differences between them.